src/ParametricOptInterface.jl

```diff
@@ -70,6 +72,191 @@ mutable struct ParametricOptimizer{T, OT <: MOI.ModelLike} <: MOI.AbstractOptimi
+    model::ParametricOptimizer,
+    F::Type{MOI.ScalarQuadraticFunction{T}},
+    S::Type{<:MOI.AbstractSet}) where T
+
+    return MOI.supports_constraint(model.optimizer, MOI.ScalarAffineFunction{T}, S)
+end
+
+function MOI.supports(
+    model::ParametricOptimizer,
+    attr::Union{MOI.ObjectiveSense,
+            MOI.ObjectiveFunction{MOI.SingleVariable},
+            MOI.ObjectiveFunction{MOI.ScalarAffineFunction{T}},
+            }) where T
+    return MOI.supports(model.optimizer, attr)
+end
+
+function MOI.empty!(model::ParametricOptimizer{T}) where T
+    MOI.empty!(model.optimizer)
+    empty!(model.parameters)
+    empty!(model.parameters_name)
+    empty!(model.updated_parameters)
+    empty!(model.variables)
+    model.last_index_added = 0
+    empty!(model.affine_constraint_cache)
+    empty!(model.quadratic_constraint_cache_pv)
+    empty!(model.quadratic_constraint_cache_pp)
+    empty!(model.quadratic_constraint_cache_pc)
+    empty!(model.quadratic_constraint_variables_associated_to_parameters_cache)
+    empty!(model.quadratic_added_cache)
+    model.last_quad_add_added = 0
+    empty!(model.affine_objective_cache)
+    empty!(model.quadratic_objective_cache_pv)
+    empty!(model.quadratic_objective_cache_pp)
+    empty!(model.quadratic_objective_cache_pc)
+    empty!(model.quadratic_objective_variables_associated_to_parameters_cache)
+    return
+end
+
+function MOI.set(model::ParametricOptimizer, attr::MOI.VariableName, v::MOI.VariableIndex, name::String)
+    if haskey(model.parameters, v)
+        model.parameters_name[v] = name
+    else
+        return MOI.set(model.optimizer, attr, v, name)
+    end 
+end
+
+function MOI.get(model::ParametricOptimizer, attr::MOI.VariableName, v::MOI.VariableIndex)
+    if haskey(model.parameters, v)
+        return model.parameters_name[v]
+    else
+        return MOI.get(model.optimizer, attr, v)
+    end
+end
+
+function MOI.supports(model::ParametricOptimizer, attr::MOI.VariableName, tp::Type{MOI.VariableIndex})
+    MOI.supports(model.optimizer, attr, tp)
+end
+
+function MOI.set(model::ParametricOptimizer, attr::MOI.ConstraintName, c::MOI.ConstraintIndex, name::String)
+    MOI.set(model.optimizer, attr, c, name)
+end
+
+function MOI.get(model::ParametricOptimizer, attr::MOI.ConstraintName, c::MOI.ConstraintIndex)
+    MOI.get(model.optimizer, attr, c)
+end
+
+function MOI.supports(model::ParametricOptimizer, attr::MOI.ConstraintName, tp::Type{MOI.ConstraintIndex})
+    MOI.supports(model.optimizer, attr, tp)
+end
+
+struct ParameterValue <: MOI.AbstractVariableAttribute end
+
+function MOI.set(model::ParametricOptimizer, ::ParameterValue, vi::MOI.VariableIndex, val)
+    cv = MOI.ConstraintIndex{MOI.SingleVariable, POI.Parameter}(vi.value)
+    MOI.set(model, MOI.ConstraintSet(), cv, POI.Parameter(val))
+end 
+
+function MOI.get(model::ParametricOptimizer, attr::MOI.ConstraintFunction, ci::MOI.ConstraintIndex{F, S}) where {F, S}
+    MOI.get(model.optimizer, attr, ci)
```

Comment:
This is not correct, you need to put the parameters back into the function. You can leave this for a later PR if you want by leaving a TODO or removing this method or commenting it out with a TODO

Reply:
@blegat, can you clarify this point? I am working on #59 that needs this method to pass

Reply:
The function returned by `model.optimizer` will contain the `Float64` value instead of the `VariableIndex` value of the parameter.